src/openms/source/FORMAT/MzTab.cpp

```diff
@@ -2179,15 +2128,68 @@ namespace OpenMS
         // we do not want descriptions twice
         protein_hit_user_value_keys.erase("Description");
 
+        // map (indist.)protein groups to their protein hits (by index).
+        //TODO if you know the number of groups it could just be a vector
+        map<Size, set<Size>> ind2prot; // indistinguishable protein groups
+        map<Size, set<Size>> pg2prot; // general protein groups
+
+        const std::vector<ProteinHit>& proteins = protein_hits;
+
+        // map indistinguishable groups to the contained proteins
+        // TODO make function. since it is the same for both group types
+        Size ind_idx{0};
+        for (const ProteinIdentification::ProteinGroup &p : indist_groups2)
+        {
+          for (const String &a : p.accessions)
+          {
+            // find protein corresponding to accession stored in group
+            auto it = std::find_if(proteins.begin(), proteins.end(), [&a](const ProteinHit &ph)
+                                   {
+                                     return ph.getAccession() == a;
+                                   }
+            );
+            if (it == proteins.end())
+            {
+              continue;
+            }
+            Size protein_index = std::distance(proteins.begin(), it);
+            ind2prot[ind_idx].insert(protein_index);
+          }
+          ++ind_idx;
+        }
 
+        // map general protein groups to the contained proteins
         // We only report quantitative data for indistinguishable groups (which may be composed of single proteins).
         // We skip the more extensive reporting of general groups with complex shared peptide relations.
-        std::vector<ProteinIdentification::ProteinGroup> protein_groups2;
+        std::vector<ProteinIdentification::ProteinGroup> empty_protein_groups;
+        auto protein_groups = std::cref(empty_protein_groups);
         if (quant_study_variables == 0)
         {
-          protein_groups2 = it->getProteinGroups();
+          protein_groups = std::cref(it->getProteinGroups());
```

Comment:
is there a difference to const std::vector<ProteinIdentification::ProteinGroup>& protein_groups2 = quant_study_variables_ == 0 ? pid.getProteinGroups() : std::vector<ProteinIdentification::ProteinGroup>();
?